dalle2_pytorch/train_diffusion_prior.py

```diff
@@ -0,0 +1,160 @@
+            if torch.cuda.is_available():
+                emb_images[0] = torch.tensor(emb_images[0]).to(device)
+                emb_text[0] = torch.tensor(emb_text[0]).to(device)
+            optimizer.zero_grad()
+            loss = diffusion_prior(text_embed = emb_text[0],image_embed = emb_images[0])
+            loss.backward()
+            # Log to wandb
+            wandb.log({"Training (mse)": loss})
+            optimizer.step()
+            print("Training mse = ",loss.item())
+            train_loss+=loss.item()
+
+        print("Validation loop - epoch number ",e)
+        with torch.no_grad():
+            valid_loss = 0.0
+            val_set_size = int(val_percent*image_reader.count)
+            start = train_set_size
+            end=start+val_set_size
+            for emb_images,emb_text in zip(image_reader(batch_size=batch_size, start=start, end=end),
+                    text_reader(batch_size=batch_size, start=start, end=end)):
+                emb_images = list(emb_images)
+                emb_text = list(emb_text)
+                if torch.cuda.is_available():
+                    emb_images[0] = torch.tensor(emb_images[0]).to(device)
+                    emb_text[0] = torch.tensor(emb_text[0]).to(device)
+                diffusion_prior.eval()
+                loss = diffusion_prior(text_embed = emb_text[0],image_embed = emb_images[0])
+
+                # Log to wandb
+                wandb.log({"Validation Loss(mse) ": loss})
+                valid_loss+=loss.item()
+
+        # Saving State Dict
+        torch.save(diffusion_prior.state_dict(), 'saved_model.pth')
+
+    test_set_size = int(test_percent*train_set_size)
+    with torch.no_grad():
+        for emb_images,emb_text in zip(image_reader(batch_size=batch_size, start=train_set_size+val_set_size,
+            end=text_reader.count),
+            text_reader(batch_size=batch_size, start=train_set_size+val_set_size, end=text_reader.count)):
+            emb_images = list(emb_images)
+            emb_text = list(emb_text)
+            if torch.cuda.is_available():
+                emb_images[0] = torch.tensor(emb_images[0]).to(device)
+                emb_text[0] = torch.tensor(emb_text[0]).to(device)
+            diffusion_prior.eval()
+            loss = diffusion_prior(text_embed = emb_text[0],image_embed = emb_images[0])
+
+            # Log to wandb
+            wandb.log({"Test mse ": loss})
+
+def main():
+    parser = argparse.ArgumentParser()
+    # Logging
+    parser.add_argument("--wandb-entity", type=str, default="laion")
+    parser.add_argument("--wandb-project", type=str, default="diffusion-prior")
+    # URLs for embeddings 
+    parser.add_argument("--image-embed-url", type=str, default="https://mystic.the-eye.eu/public/AI/cah/laion5b/embeddings/laion2B-en/img_emb/")
+    parser.add_argument("--text-embed-url", type=str, default="s3://laion-us-east-1/embeddings/vit-l-14/laion2B-en/text_emb/")
```

Comment:
can you point to https://mystic.the-eye.eu/public/AI/cah/laion5b/embeddings/laion2B-en/text_emb/ instead so that everyone has access ?

Reply:
Fixing this right away.